hashicorp/terraform
# Cloud: Make e2e Tests Less Chatty, and More Stable

## Description

This fixes the Cloud e2e tests.

**Changes**
* Run table tests as subtests (in parallel)
* Fix broken tests
* Remove `TF_LOG=INFO` in test cases
* Add a readme to show you to run tests.
* Add new `-tfoutput` flag to print the terraform to standard output.
* Change expect to not log to STDOUT
* Some commands were changed to use `-auto-approve` because it wasn't necessary for extra input steps.

_Note:_ I've removed some `t.Parallel()` bits because it started introducing some flakiness. I didn't do a full investigation because of time considerations. We will investigate this at a later point.

### Test Output 

```
terraform % TF_ACC=1 go test ./internal/cloud/e2e/... -v -ldflags "-X \"github.com/hashicorp/terraform/version.Prerelease=alpha20211029\"" 
...
...
PASS
ok      github.com/hashicorp/terraform/internal/cloud/e2e       489.176s
```
